From 0693729fc4d89fe937d2f975ab8ce28c3a44139e Mon Sep 17 00:00:00 2001
From: Simon McVittie <smcv@debian.org>
Date: Fri, 20 Apr 2018 09:09:52 +0100
Subject: [PATCH 3/4] qemu: Wait for shell prompt before flag file

Waiting for the commands to complete on the qemu side seems to be
a bit more reliable than waiting for the flag file on the host side.

Signed-off-by: Simon McVittie <smcv@debian.org>
---
 virt/autopkgtest-virt-qemu | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/virt/autopkgtest-virt-qemu b/virt/autopkgtest-virt-qemu
index 6f3fef2..89efaea 100755
--- a/virt/autopkgtest-virt-qemu
+++ b/virt/autopkgtest-virt-qemu
@@ -225,12 +225,12 @@ mount -t 9p -o trans=virtio,access=any autopkgtest /run/autopkgtest/shared &&
 chmod 1777 /run/autopkgtest/shared &&
 touch /run/autopkgtest/shared/done_shared
 ''')
+    VirtSubproc.expect(term, b'#', 30)
 
     with VirtSubproc.timeout(10, 'timed out on client shared directory setup'):
         flag = os.path.join(shared_dir, 'done_shared')
         while not os.path.exists(flag):
             time.sleep(0.2)
-    VirtSubproc.expect(term, b'#', 30)
 
     # ensure that root has $HOME set
     term.send(b'[ -n "$HOME" ] || export HOME=`getent passwd root|cut -f6 -d:`\n')
-- 
2.17.0

